Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+
+final class DefaultBillingFormViewModel: BillingFormViewModel {
+    var style: BillingFormStyle
+    var updateRow: (() -> Void)?
+    var updatedRow: Int? {
+        didSet {
+            updateRow?()
+        }
+    }
+    
+    lazy var errorFlagOfCellType = [BillingFormCellType: Bool]()
+    lazy var textValueOfCellType = [BillingFormCellType: String]()
+    
+    private var initialCountry: String
+    private var initialRegionCode: String?
+    var editDelegate: BillingFormViewModelEditingDelegate?
+    var delegate: BillingFormViewModelDelegate?
+    
+    init(style: BillingFormStyle, initialCountry: String = "", initialRegionCode: String? = nil, delegate: BillingFormViewModelDelegate? = nil) {
+        self.style = style
+        self.initialCountry = initialCountry
+        self.initialRegionCode = initialRegionCode
+        self.delegate = delegate
+    }
+    
+    func getCell(for row: Int, delegate: UIViewController? = nil) -> UITableViewCell {
+        if row == 0 {
+            return getHeaderCell(delegate: delegate as? BillingFormHeaderCellDelegate)
+        }
+        return getFieldCell(for: row, delegate: delegate as? BillingFormTextFieldCellDelegate)
+    }
+     
+    private func getHeaderCell(delegate: BillingFormHeaderCellDelegate?) -> UITableViewCell {
+        var style = style.header
+        updateHeaderView(with: &style)
+        let cell = BillingFormHeaderCell(style: style, delegate: delegate)
+        self.editDelegate = cell
+        return cell
+    }
+    
+    private func getFieldCell(for row: Int, delegate: BillingFormTextFieldCellDelegate?) -> UITableViewCell {
+        guard let type = BillingFormCellType(rawValue: row) else { return UITableViewCell()}
+        var style = style.fields[row - 1]
+        updateErrorView(with: &style, type: type)
+        updateTextField(with: &style, type: type)
+        return BillingFormTextFieldCell(type: type, style: style, delegate: delegate)
```

Comment:
I am afraid your current approach is missing cell dequeuing. As a practice this is something I disagree with, but I am not sure I understand sufficiently to assess if you found an edge case where this becomes acceptable? Happy to understand more.

Reply:
Your approach to the use of style is very interesting. You start with a default value, then attempt to override a number of times, whilst the methods are called `update....` and yet they don't update what they claim to, they just mutate a property. 

Can we rethink this flow ti a more common way?